darylldoyle/safe-svg
# Add a check to ensure we have a valid PHP version

### Description of the Change

The minimum PHP version was bumped in #82 which went out in the v2.1.0 release. We've had a few reports of users running a version of PHP less than 7.4 that have somehow been able to update to the latest version of the plugin (ideally WordPress would prevent this but that's another issue to solve).

Because of the new PHP version we set in our `composer.json` file, composer is throwing a fatal error in those situations. Ideally we should never throw fatal errors, so this PR fixes this by adding a PHP version check that renders an admin error message and disables the rest of the plugin from functioning.

<img width="712" alt="Screen Shot 2023-03-24 at 11 00 54 AM" src="https://user-images.githubusercontent.com/916738/227592100-f9d6afd8-a416-45a4-9644-2409600a18be.png">

Closes #110 

### How to test the Change

1. Setup a WordPress environment running PHP 7.3
2. Install the v2.1.0 of Safe SVG and see a fatal error happen
3. Update Safe SVG to this branch and note no fatal error should happen but an admin error message should show

### Changelog Entry

> Fixed - Better error handling for environments that don't match our minimum PHP version


### Credits

Props @dkotter


### Checklist:
